src/DeFuncto.Tests/Core/Types/AsyncResult/Linq.cs

```diff
@@ -50,4 +50,51 @@ from ok7 in Ok<string, int>(f.Get).Async()
 
         int Boom() => throw new Exception("Should not happen");
     }
+
+    [Property(DisplayName = "AsyncResult Select should project results")]
+    public void SelectProjectsResult()
+    {
+        // leave the type castings to ensure it's not returning a nested Result<Result...>>
+
+        ((AsyncResult<decimal, int>)(from _ in Ok<string, int>(string.Empty).Async() select Ok<decimal, int>(decimal.Zero).Async())).ShouldBeOk();
+        ((AsyncResult<string, int>)(from _ in Ok<string, int>(string.Empty).Async() select Ok<string, int>(string.Empty).Async())).ShouldBeOk();
+
+        ((AsyncResult<string, int>)(from _ in Ok<string, int>(string.Empty).Async() select Error<string, int>(1).Async())).ShouldBeError(1);
+        ((AsyncResult<string, int>)(from _ in Error<string, int>(1).Async() select Error<string, int>(2).Async())).ShouldBeError(1);
+        ((AsyncResult<string, int>)(from _ in Error<string, int>(1).Async() select Ok<string, int>(string.Empty).Async())).ShouldBeError(1);
+    }
+
+    [Property(DisplayName = "AsyncResult SelectMany should project results")]
+    public void SelectManyProjectsResult()
```

Comment:
This will never fail if there's a runtime error. You have to await it to be sure.